core/templates/pages/exploration-editor-page/services/version-history.service.ts

```diff
@@ -22,33 +22,33 @@ import { ExplorationMetadata } from 'domain/exploration/ExplorationMetadataObjec
 import { State } from 'domain/state/StateObjectFactory';
 
 export interface StateDiffData {
-  oldState: State;
-  newState: State;
-  oldVersionNumber: number;
-  newVersionNumber: number;
+  oldState: State | null;
```

Comment:
Just a general note: I'm actually getting quite concerned about the propagation of this "| null". We should try to minimize that where possible, otherwise it leads to uncertainty in the types. Can you justify why all these are needed and why we can't avoid them?

Reply:
I have added some explanations and also removed "| null" from many places. PTAL. Thanks!